libraries/AP_HAL_ChibiOS/hwdef/StellarF4V2/hwdef.dat

```diff
@@ -0,0 +1,153 @@
+PC2  SPI2_MISO SPI2
+PB15 SPI2_MOSI SPI2
+PB12 MAX7456_CS CS
+
+# SPI3 flash
+PB3  SPI3_SCK SPI3
+PB4  SPI3_MISO SPI3
+PB5  SPI3_MOSI SPI3
+PC13 FLASH_CS CS
+
+# only one I2C bus in normal config
+PB10 I2C2_SCL I2C2
+PB11 I2C2_SDA I2C2
+
+# ADC
+PC0 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+PC1 BATT_CURRENT_SENS ADC1 SCALE(1)
+
+# define default battery setup
+define HAL_BATT_MONITOR_DEFAULT 4
+define HAL_BATT_VOLT_PIN 10
+define HAL_BATT_CURR_PIN 11
+define HAL_BATT_VOLT_SCALE 11.0
+define HAL_BATT_CURR_SCALE 10.0
+
+# USART2 (RCIN/SBUS)
+PA2 USART2_TX USART2
+PA3 USART2_RX USART2
+define DEFAULT_SERIAL1_PROTOCOL SerialProtocol_RCIN
+
+# USART3
+PC10 USART3_TX USART3 NODMA
+PC11 USART3_RX USART3 NODMA
+define DEFAULT_SERIAL2_PROTOCOL SerialProtocol_None
+
+# UART4 (DJI)
+PA0 UART4_TX UART4 NODMA
+PA1 UART4_RX UART4 NODMA
+define DEFAULT_SERIAL3_PROTOCOL SerialProtocol_MSP_DisplayPort
+
+# UART5 (ESC Telemetry)
+PC12 UART5_TX UART5 NODMA
+PD2  UART5_RX UART5 NODMA
+define DEFAULT_SERIAL4_PROTOCOL SerialProtocol_ESCTelemetry
+
+# PWM out pins. Note that channel order follows the ArduPilot motor
+# order conventions
+PC9  TIM8_CH4 TIM8 PWM(1)  GPIO(50) BIDIR # M1
+PC8  TIM8_CH3 TIM8 PWM(2)  GPIO(51)       # M2
+PC7  TIM8_CH2 TIM8 PWM(3)  GPIO(52) BIDIR # M3
+PC6  TIM8_CH1 TIM8 PWM(4)  GPIO(53)       # M4
+PB6  TIM4_CH1 TIM4 PWM(5)  GPIO(54)       # S1M5
+PB7  TIM4_CH2 TIM4 PWM(6)  GPIO(55)       # S2M6
+PB8  TIM4_CH3 TIM4 PWM(7)  GPIO(56)       # S3M7
+PB9  TIM4_CH4 TIM4 PWM(8)  GPIO(57)       # S4M8
+PA9  TIM1_CH2 TIM1 PWM(9)  GPIO(58)       # S5
+PA10 TIM1_CH3 TIM1 PWM(10) GPIO(59)       # S6
+PA8  TIM1_CH1 TIM1 PWM(11) GPIO(60)       # S7
+
+
+# reserve 16k for bootloader and 32k for flash storage
+FLASH_RESERVE_START_KB 48
+
+# SPI devices
+SPIDEV icm42688  SPI1 DEVID1 GYRO1_CS MODE3 2*MHZ 16*MHZ
+SPIDEV osd       SPI2 DEVID1 MAX7456_CS MODE0 10*MHZ 10*MHZ
+SPIDEV dataflash SPI3 DEVID2 FLASH_CS  MODE0 500*KHZ 25*MHZ
+
+# IMU icm42688
+IMU Invensensev3 SPI:icm42688 ROTATION_PITCH_180
+IMU BMI270 SPI:bmi270 ROTATION_PITCH_180
+
+# built-in barometer
+BARO SPL06  I2C:0:0x76
+BARO DPS310 I2C:0:0x76
+BARO BMP280 I2C:0:0x76
+
+# no built-in compass, but probe the i2c bus for all possible
+# external compass types
+define ALLOW_ARM_NO_COMPASS
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+define HAL_I2C_INTERNAL_MASK 0
+
+# Motor order implies Betaflight/X for standard ESCs
+define HAL_FRAME_TYPE_DEFAULT 12
+
+# enable logging to dataflash
+define HAL_LOGGING_DATAFLASH_ENABLED 1
+define HAL_STORAGE_SIZE 15360
+STORAGE_FLASH_PAGE 1
+
+# setup for OSD
+define OSD_ENABLED 1
+define HAL_OSD_TYPE_DEFAULT 1
+ROMFS_WILDCARD libraries/AP_OSD/fonts/font*.bin
+
+# minimal drivers to reduce flash usage
+include ../include/minimize_fpv_osd.inc
+include ../include/no_bootloader_DFU.inc
+undef AP_BATTERY_SYNTHETIC_CURRENT_ENABLED
+define AP_BATTERY_SYNTHETIC_CURRENT_ENABLED 0
+
+define DEFAULT_NTF_LED_TYPES 257
+
+# GPIOs 81:CAM SELECT
+PC14  PINIO1 OUTPUT GPIO(81) LOW
```

Comment:
its convenient to set this up as a relay so that a user can easily put it on an rc switch